src/syncerd/types.rs

```diff
@@ -316,29 +356,64 @@ pub struct TaskAborted {
     pub error: Option<String>,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
-#[display(Debug)]
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 pub struct HeightChanged {
     pub id: TaskId,
     pub block: Vec<u8>,
     pub height: u64,
 }
 
+impl fmt::Display for HeightChanged {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "HeightChanged(id: {}, block: {}, height: {})",
+            self.id,
+            hex::encode(&self.block),
+            self.height,
+        )
+    }
+}
+
 #[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
```

Comment:
`Copy` should be derived here too as both `monero::Hash` and `bitcoin::Txid` are `Copy`

Reply:
3ad79ed808eb9aa7e58158514c1dffb77da03380